### ISSUE TITLE
Weather monitor no longer working in background

### ISSUE THREAD

**Issue by intr-cx** [kylecorry31/Trail-Sense#964]
The weather graph and forecast no longer update when the app is not focused. It appears the graph only gets new points when opening the app.
![screenshot](https://i.imgur.com/QvW2Atg.png)
You can basically see exactly when I opened the app to check.

I use GrapheneOS with the (as of writing) latest F-Droid version 2.9.0. I have explicitly disabled any battery optimization settings, to no avail. Before 2.9.0 it worked just fine.

**Comment by kylecorry31** [comment]
I'll also be posting a debug / beta build before I publish the next release (https://github.com/kylecorry31/Trail-Sense/issues/74) if you are interested in seeing if it is fixed (the setting will be under Experimental > Background compatibility mode, though I might change that eventually)

**Comment by kylecorry31** [comment]
Turns out that the barometer can't be read in the background without the compatibility mode enabled (foreground service): https://developer.android.com/guide/topics/sensors/sensors_overview#only-gather-sensor-data-in-the-foreground

I've turned on the "compatibility" mode by default and removed the setting.

I still have no idea why it works fine on my device though... (running Android 11)

I've added this issue to track Android 12 support: #971. It will work for now until I actually target Android 12 which won't be required until late next year.
